src/docfx/lib/msgraph/MicrosoftGraphAuthenticationProvider.cs

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Graph;
+using Microsoft.Identity.Client;
+
+namespace Microsoft.Docs.Build
+{
+    internal class MicrosoftGraphAuthenticationProvider : IAuthenticationProvider, IDisposable
+    {
+        private static string resource = "https://graph.microsoft.com/.default";
+        private static string[] scopes = new string[] { resource };
+
+        private readonly ConfidentialClientApplication cca;
+        private AuthenticationResult authenticationResult;
```

Comment:
```suggestion
        private readonly AuthenticationResult authenticationResult;
```

Reply:
authenticationResult is assigned in function: GetAccessTokenAsync , so we can't set it readonly